src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartContainer.java

```diff
@@ -1,85 +1,164 @@
 package com.hivemq.helmcharts.util;
 
+import com.github.dockerjava.api.DockerClient;
+import com.hivemq.crd.hivemq.HiveMQInfo;
+import com.hivemq.openapi.HivemqClusterStatus;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.WatcherException;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
-import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.startupcheck.StartupCheckStrategy;
 import org.testcontainers.images.builder.ImageFromDockerfile;
 import org.testcontainers.k3s.K3sContainer;
 import org.testcontainers.utility.DockerImageName;
 import org.testcontainers.utility.MountableFile;
 
 import java.io.File;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.testcontainers.containers.output.OutputFrame.OutputType.STDERR;
 
 /**
  * Container that includes the helm binary to be able to install the HiveMQ helm charts
  */
 public class OperatorHelmChartContainer extends K3sContainer {
     public final static int mqttPort = 1883;
+    private final @NotNull List<String> imagesNames;
+    private boolean withCustomImages = false;
 
-    private OperatorHelmChartContainer(final @NotNull String k3sVersion,
-                                       final @NotNull File dockerfile,
-                                       final @NotNull File helmChartMountPath,
-                                       final @NotNull String containerPath) {
-        super(getAdHocImageName(k3sVersion, dockerfile));
-        final MountableFile helmChartMountable = MountableFile.forHostPath(helmChartMountPath.getPath());
-        super.withCopyFileToContainer(helmChartMountable, containerPath);
+    public OperatorHelmChartContainer(final @NotNull String k3sVersion,
+                                      final @NotNull String dockerfileName,
+                                      final @NotNull String customValuesFile) {
+        super(getAdHocImageName(k3sVersion, dockerfileName));
         super.addExposedPort(mqttPort);
+        super.withFileSystemBind("./charts/hivemq-operator", "/chart");
```

Comment:
Use `withCopyFileToContainer` instead of `withFileSystemBind`. `withCopyFileToContainer` seems to be more stable on CIs.